### PR TITLE
Sleeping Carp: Scaled Skin Edition

### PR DESCRIPTION
**Pull request by necromanceranne**
## About The Pull Request

So I figured it was about time that Sleeping Carp was reworked, but I really struggled to figure out a good middle ground for power and where to take it. I had a few goals in mind. Limited by my...really limited ability to code:

**One:** Removing the hardstuns from the martial art.
**Two:** Emphasis on what carp was being take for that wasn't hard stuns.
**Three:** Try and give every single one of the moves a use.
**Four:** Make up for the loss of hard stuns, which was what was keeping the martial art functional.

As a result, I've created a slightly different kind of Sleeping Carp. It's features include:

**Conditionally Immune to Projectiles:** Since people were very keen for this to remain, I've added ranged immunity while you have Throw Mode up and only then. So you can't deflect ranged attacks and attack at the same time. More important for the approach.

**Now Heavily Resistant to Near All Damage and Disables:** Basically, the Sleeping Carp user is a tough fucking bastard of a martial artist. They have resistance to most every kind of damage except Oxygen, Clone and Fire/Heat. They are able to spacewalk temporarily as a result, given they take very little damage from pressure, cold and brute and burn. They're more akin to a carp than man. Or skeleton. Or lightbulb. Or moth. Whatever.

**Higher Vulnerability to Things They Weren't Vulnerable too Before:** You can shoot them with guns and it will now work, **mostly**. Which means Joe Shmoe, Security Officer fresh off the boat, isn't reliant on specialized equipment to fight a sleeping carp user. He's probably still in deep shit if he's fighting them on his own, however. Temp Guns set to bake will work well here too.

**Immunity to Relevant Stuff, Stronger with Armor:** If your skin is as thick as scales, then things that could puncture that no longer work. They are immune to syringes, sutures, ninja stars (I think, embed is weird), embedding spears (take a little, give a little). In addition, armor becomes way more important for Sleeping Carp users to encourage them to really focus in on the defense aspect of their martial art and tank damage.

**Carp Faction:** Carp petting.

**Only Three Moves:** All the moves are started with Harm intent and ended with one of the three other intents for whatever you aim to do. These are:
**Gnashing Teeth:** _One-Two punch with a potential for a critical hit. Basic damage move._
**Crashing Wave Kick:** _A kick that launches people off the screen. Good for keeping anyone off you and really fucking them up in space._
**Keelhaul:** _Dropkick that knocks down and disarms. While the person is on the floor, disarms and applies stamina damage. Pretty much your main opener._

**Increased Subtly:** The only way someone will know you have this is if they try and use anything that requires you not to be in a hardsuit. Sutures, syringes, etc. Otherwise the effects are immensely subtle until you start punching people.

I've removed aggressive grabs for the time being. I didn't think it would be entirely relevant but I can understand people liked that part for body throws. There is some of this still in the martial art in spirit with launch kicking people into each other. But obviously that isn't as accurate as body throwing.

In addition to all these changes, I've changed batons to deal apply_damage instead of adjustStaminaLoss. This SHOULDN'T affect anything other than making it so that stamina_mod actually is relevant against stun batons. And maybe limb targeting being a thing with batons.

## Why It's Good For The Game

Sleeping Carp wasn't going to magically work by removing the stuns. It was straight up reliant on the stuns to keep people locked into place in order to perform the sequence of moves. If it didn't have stuns and aggressive grab, the moves would be pretty much impossible to perform. I hopefully have found a good counterbalance that makes the user of Sleeping Carp still a powerful melee combatant and able to mulch face, but not nearly as over reliant on stuns to achieve this. 

My goal is for it to feel more like you're beating the shit out of your opponent then them standing there helplessly while you apply damage.

## Changelog
:cl: NecromancerAnne
tweak: Completely reworked Sleeping Carp
balance: Sleeping Carp now costs 15TC.
balance: Sleeping Carp users are really goddamn durable.
balance: Sleeping Carp users have less moves but shorter combos and more ultraviolence.
balance: Carp faction. Space walking-lite.
balance: Sleeping Carp users need to be in throw mode to deflect ranged attacks.
balance: Sleeping Carp users are immune to objects that puncture the skin.
balance: Stun batons are now apply_damage instead of applyStaminaLoss. This might change some things.
/:cl:

### CODE REVIEW

**Comment by Tlaltecuhtli** [comment]
how many slug in head does he need now to crit

**Comment by Archanial** [comment]
Does carp really need a rework? It still costs 17 tc and you still are somewhat a glass cannon. 
I get it - stuns are annoying but traitors are supposed to be deadly.
With all the changes made to carp now it looks like a worse ling.

**Comment by necromanceranne** [comment]
A price adjustment is something I've been thinking about too. I don't think it's quite worth 17tc anymore. It feels lower.

@Tlaltecuhtli try fighting one and find out. Presumably about three more or so? But that's shooting into someone without armor.

**Comment by Fikou** [comment]
should have some ability to approach people with confidence, a move to mute someone

**Comment by Archanial** [comment]
If you're going to reduce price, what about gloves of the north star and carp interaction?

**Comment by necromanceranne** [comment]
If kept in the 13-14 tc range it's still overpriced for the combo without discounts. Other thanth. Thisis an issue pretty frequently brought up about GotNS with any martial art and changes with that probably need to be addressed separately but it is something to be worried about. Especially since it combos fairly powerfully together with the new Sleeping Carp having critical hits on Gnashing Teeth. It doesn't have paralysis punches anymore, however, so you can't stunlock with punches alone.

**Comment by AWalton** [comment]
There's reworks and then there's creating a completely different thing. Add this as a new separate item and later on if desired do a separate PR to remove the old martial art. Trying to use the name Sleeping Carp for this feels like trying to ride off the brand name to get a new martial art merged.

**Comment by Angustmeta** [comment]
Not liking the RNG crit chance. The crit could be used as a result of a combo instead.

**Comment by GuillaumePrata** [comment]
I really like the rework idea, but as said above using the name Sleeping Carp is weird, Sleeping Carp was always about the dumb immunity to projectiles for me, its what made it different of CQC and removing it while keeping the name feels wrong.

Removing the stuns would be great tho, its fun to be immune to projectiles and feel like Darth Vader walking the halls and not caring about their puny laser, maybe a chance to reflect projectile back to the user would be a fair trade along with a possible price drop.

I would use another name and keep both arts, possibly related to space sharks since they are the tougher carp basically while Sleeping is a more agile style. Also a Dragon fighting style in the future just to complete the Carp Family Martial arts maybe...

**Comment by itseasytosee** [comment]
>They are immune to syringes, sutures, ninja stars

**Comment by wesoda25** [comment]
Crashing wave kick sounds op, but also fun as fuck and I really hope it gets added

**Comment by dr-bee** [comment]
I can see this getting stupid when used with augmentation, armor nanites, and armor, due to the way armor and damage reduction interact.

Maybe instead of a straight percentile reduction in damage have it be innate armor so armor piercing works?

**Comment by HomocidlePancakes** [comment]
1)Doesn’t need a rework.
2)New meta bois. Aim for the legs with the baton to slow people to a grinding halt that 100% guarantees a second hit.

**Comment by necromanceranne** [comment]
Species armor actually isn't much better and isn't discriminatory about damage reduction.

**Comment by SkeletalElite** [comment]
Does the stun baton change mean stun batons respect armor? Which armor type does it respect if so. Also disabling peoples legs and arms with the baton will be interesting.

**Comment by ATH1909** [comment]
what if you kept the 100% projectile immunity, but only made it work while the sleeper carp user is in throw mode? that way, you could still use it to close the gap to someone, but you can't have your projectile immunity and punch people at the same time?

**Comment by necromanceranne** [comment]
@ATH1909 that's not a bad suggestion. 

I am beginning to see people quite liked the ranged immunity, so I think I might try and re-add it somewhere, somehow.

**Comment by necromanceranne** [comment]
I've readded bullet deflection. So now, while in Throw Mode, you gain ranged immunity and also 40% block chance. This obviously comes with martial counter, much like CQC, but it isn't nearly as strong for obvious reasons.

**Comment by necromanceranne** [comment]
@SkeletalElite Oh also sorry to mention this. No it shouldn't be respecting armor (at the moment). And frankly it wouldn't be any different to just hitting someone in the chest. You could disarm more easily and the loss of a leg will make it harder to run, Who knows.

**Comment by ATH1909** [comment]
> @SkeletalElite Oh also sorry to not mention this. No it shouldn't be respecting armor (at the moment). And frankly it wouldn't be any different to just hitting someone in the chest. You could disarm more easily and the loss of a leg will make it harder to run, Who knows.

Damn, am I finally gonna have to set up arm targetting hotkeys?

Can a single stun baton hit force someone to instantly drop something due to the stamina damage on their arm?

**Comment by necromanceranne** [comment]
Well my waa tested and wouldn't you know, it's still applying to the chest. So it will function as normal right night.

The real question is, do people actually want limb targeting batons?

Also, another fun fact, so is sleeping carp probably. And has always been doing so. Which is weird. So I guess I'll fix that.

**Comment by necromanceranne** [comment]
I took out the block after all. It was too easy to just sit behind it and tank and win without any real effort. Martial Counter is kind of a shit mechanic in of itself that probably needs a big re-evaluation but  for now we'll just take out the block and I'll consider it down the line.

**Comment by ATH1909** [comment]
> I took out the block after all. It was too easy to just sit behind it and tank and win without any real effort. Martial Counter is kind of a shit mechanic in of itself that probably needs a big re-evaluation but for now we'll just take out the block and I'll consider it down the line.

Well, if sec is emptying all of their ammo into you while you're standing still and obviously deflecting all of their shots, I feel like that's sort of on them, not on you. You can also throw an item at the sleeping carp user to take them off of throw mode, then get a hit or two in with a ranged weapon before they can react and reenable throw mode.

You yourself said that you wanted this martial art to emphasize defense over offense, and I feel like the throwmode ammo stalling tactic is totally in that theme. Also, remember that you ARE spending 17 TC on learning this martial art and that double-bladed eswords also block lasers (and some bullets) and cost less than a sleeping carp scroll does (and doesn't permanently forbid you from using ranged weapons!).

**Comment by necromanceranne** [comment]
They still have bullet deflection just not baton counter.

**Comment by AWalton** [comment]
It's a bit cringe that an item that's supposed to make you godly in melee combat can still end up with you getting dabbed on by a baton user. Before you could hit them once with an aggro grab to stunlock punch to death, now you've got to risk doing two hits to finish a combo, and even more on top of that to kill them.

**Comment by necromanceranne** [comment]
You have like four-five baton hits to take them out. If you still can't win then maybe the issue isn't with the martial art at that point but stun batons being obscenely more powerful than any other melee weapon available.

**Comment by necromanceranne** [comment]
I reduced the cost of sleeping carp to 15TC. I chose this number because people had some concerns with emag and sleeping carp being too much of an obvious purchase combination. This at least lets you have more room to fit a few devices you need to do what you need to do without the infinite value of emags on top of this, since this has already an immense amount of value in of itself. 

Things like the syndicate bible, and lots of carp friends. :3

I'll consider this to be a similar balancing price reduction to what ebow was given, and that proved to be a smart choice in the end since it was refocused away from being such a surefire means of killing people.

**Comment by Ty-the-Smonk** [comment]
Scroll should come with the carp suit

File: code/datums/martial/sleeping_carp.dm [hunk 1]
```diff
@@ -162,22 +105,45 @@
...
-
+	REMOVE_TRAIT(H, TRAIT_PIERCEIMMUNE, SLEEPING_CARP_TRAIT)
+	REMOVE_TRAIT(H, TRAIT_STUNRESISTANCE, SLEEPING_CARP_TRAIT)
+	REMOVE_TRAIT(H, TRAIT_NODISMEMBER, SLEEPING_CARP_TRAIT)
+	H.physiology.brute_mod *= 2
+	H.physiology.burn_mod *= 2
+	H.physiology.stamina_mod *= 2
+	H.physiology.stun_mod *= 2
+	H.physiology.pressure_mod *= 2 //no more carpies
+	H.physiology.cold_mod *= 2
```
Comment (kriskog) [inline code/datums/martial/sleeping_carp.dm]: These need to at the very least attempt to return it to the initial damage mods.

**Review by kriskog — CHANGES_REQUESTED** [review]
Fix the thing.

Otherwise, looks fine to give this a go. It might be a bit strong, but it can be tuned down as we get data.

**Comment by adamkad1** [comment]
The 'no permanent projectile immunity' is actually good, you cant be validchecked by people with foam dart guns or wands of nothing. You can be checked by a syringe but randomly shooting people with syringes is a good way to get lynched

**Comment by Mey-Ha-Zah** [comment]
Hmmm quick question. Since sleeping carp no longer reflects projectiles, can you remove the fact that Colossus will rage target them, since the rage was to counter the ranged immunity?

**Comment by adamkad1** [comment]
> Hmmm quick question. Since sleeping carp no longer reflects projectiles, can you remove the fact that Colossus will rage target them, since the rage was to counter the ranged immunity?

You still do, just not allways

**Pull request merged**